src/server/game/Entities/Object/Object.cpp

```diff
@@ -2252,7 +2252,7 @@ TempSummon* Map::SummonCreature(uint32 entry, Position const& pos, SummonPropert
 
     summon->SetVisibleBySummonerOnly(visibleBySummonerOnly);
 
-    if (!AddToMap(summon->ToCreature(), summon->GetOwnerGUID().IsPlayer() || (summoner && summoner->GetTransport())))
+    if (!AddToMap(summon->ToCreature(), summon->GetOwnerGUID().IsPlayer() || (summoner && summoner->GetTransport() && !(summoner->IsUnit() && summoner->ToUnit()->IsVehicle()))))
```

Comment:
Maybe we can improve readability a little by introducing two variables?
```suggestion
	bool summonerHasTransport = summoner && summoner->GetTransport();
	bool summonerIsNotVehicle = !(summoner->IsUnit() && summoner->ToUnit()->IsVehicle());
    if (!AddToMap(summon->ToCreature(), summon->GetOwnerGUID().IsPlayer() || (summonerHasTransport && summonerIsNotVehicle)))
```